contracts/oracles/CurveLPOracle.sol

```diff
@@ -0,0 +1,51 @@
+// SPDX-License-Identifier: bsl-1.1
+
+/*
+  Copyright 2020 Unit Protocol: Artem Zakharov (az@unit.xyz).
+*/
+pragma solidity ^0.7.1;
+
+import "./OracleSimple.sol";
+import "../helpers/ERC20Like.sol";
+import "./OracleRegistry.sol";
+
+interface CurveProvider {
+    function get_registry() external view returns (address);
+}
+
+interface CurveRegistry {
+    function get_pool_from_lp_token(address) external view returns (address);
+    function get_virtual_price_from_lp_token(address) external view returns (uint);
+}
+
+/**
+ * @title CurveLPOracle
+ * @dev Oracle to quote curve LP tokens
+ **/
+contract CurveLPOracle is OracleSimple {
+
+    uint public constant Q112 = 2 ** 112;
+    uint public constant PRECISION = 1e18;
+
+    // CurveProvider contract
+    CurveProvider public immutable curveProvider;
+
+    /**
+     * @param _curveProvider The address of the Curve Provider. Mainnet: 0x0000000022D53366457F9d5E68Ec105046FC4383
+     **/
+    constructor(address _curveProvider) {
+        require(_curveProvider != address(0), "Unit Protocol: ZERO_ADDRESS");
+        curveProvider = CurveProvider(_curveProvider);
+    }
+
+    // returns Q112-encoded value
+    function assetToUsd(address asset, uint amount) public override view returns (uint) {
+        if (amount == 0) return 0;
+        CurveRegistry cR = CurveRegistry(curveProvider.get_registry());
+        require(cR.get_pool_from_lp_token(asset) != address(0), "Unit Protocol: NOT_A_CURVE_LP");
+
+        uint price_q112 = cR.get_virtual_price_from_lp_token(asset) * Q112 / PRECISION;
+        return amount * price_q112;
```

Comment:
What if asset's decimals are not 18? Return value should be adjusted in this case.
Or we should require decimals == 18.

Reply:
According to the [docs](https://curve.readthedocs.io/registry-registry.html#Registry.get_virtual_price_from_lp_token) this value is `normalized to a 1e18 precision`

Reply:
According to the [docs](https://curve.readthedocs.io/registry-registry.html#Registry.get_virtual_price_from_lp_token) seems this value is always normalized to 1e18. It depends on pool's `PRECISION` that equal to 1e18 even in [BTC-based pool](https://etherscan.io/address/0x93054188d876f558f4a66B2EF1d97d16eDf0895B#code)

Reply:
No-no, `PRECISION` is not relevant here as it is taken into account on line 47.
The problem on line 48 is similar to the one we had with chainlink oracle. We're multiplying price by `amount` assuming that the `decimals` of the `asset` are 18. If they're not, the usage of returned value (e.g. in `function UR`) incurs error since USDP decimals are 18.

Reply:
> Shouldn't oracle stuff go to https://github.com/unitprotocol/oracles?

Yea, some of the oralce logic is isolated in another repo due to inability to test Keydonix stuff with truffle. But these contracts are compatible with truffle, so there is no reason to isolate it.